### PR TITLE
CI: upgrade to latest libressl 3.6.1

### PR DESCRIPTION
**Pull request by frankmorgner**
##### Checklist

- [ ] Documentation is added or updated
- [ ] New files have a LGPL 2.1 license statement
- [ ] PKCS#11 module is tested
- [ ] Windows minidriver is tested
- [ ] macOS tokend is tested

### CODE REVIEW

File: .github/setup-libressl.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 set -ex -o xtrace
 
-V=libressl-3.4.2
+V=libressl-3.6.1
```
[the remark this answers is not in the document]
Comment (frankmorgner) [reply .github/setup-libressl.sh:5]: Indeed, that would be nice. However, I don't see that dependabot can be easily integrated into our build scripts.

**Comment by Jakuje** [comment]
Shall we jump to 3.7.0 already? Or add some script to check what is the latest file in the https://ftp.openbsd.org/pub/OpenBSD/LibreSSL/ directory to notify us about newest version?

**Comment by frankmorgner** [comment]
The latest stable release is 3.6.1.  The latest development release is 3.7.0. That's why I'd like to stick to 3.6.1 for the CI tests

**Comment by frankmorgner** [comment]
Yes, we're good now.

**Comment by frankmorgner** [comment]
> add some script to check what is the latest file in the https://ftp.openbsd.org/pub/OpenBSD/LibreSSL/ directory to notify us about newest version?

If you think that could be quickly implemented, I'm happy to dump the stable release for an automated solution.

**Comment by Jakuje** [comment]
I opened a new issue #2670 to keep track of this as I did not have time to look into that so far. We do not need to delay this fix so merging.

**Pull request merged**